### PR TITLE
Usability improvements for waves

### PR DESCRIPTION
**Pull request by doanac**
* require a tag name when creating a wave
* fix issue with "target collisions"
* add ability to display production targets.json
* add ability to show a production target

### CODE REVIEW

File: subcommands/targets/list.go [hunk 1]
```diff
@@ -55,21 +58,39 @@ func init() {
 	}
 	cmd.AddCommand(listCmd)
 	listCmd.Flags().BoolVarP(&listRaw, "raw", "r", false, "Print raw targets.json")
+	listCmd.Flags().BoolVarP(&listProd, "production", "", false, "Show the production version targets.json")
```
Comment (vkhoroz) [inline subcommands/targets/list.go:61]: I thought I've already added this ability.
Thank you for doing this.

**Review by vkhoroz — APPROVED** [review]
Good job!

File: client/foundries.go [hunk 2]
```diff
@@ -1048,8 +1048,11 @@ func (a *Api) TufProdRootPost(factory string, root []byte) (string, error) {
 	return a.tufRootPost(factory, true, root)
 }
 
-func (a *Api) TargetsListRaw(factory string) (*[]byte, error) {
+func (a *Api) TargetsListRaw(factory string, productionTag string) (*[]byte, error) {
 	url := a.serverUrl + "/ota/repo/" + factory + "/api/v1/user_repo/targets.json"
+	if len(productionTag) > 0 {
+		url += "?production=1&tag=" + productionTag
```
Comment (vkhoroz) [inline client/foundries.go:1054]: Oops!  We have a production targets API at this URL: "/ota/repo/" + factory + "/prod-targets/?tag=" + productionTag.

And I think it'd be good to have a `ProdTargetsList` function which works with it.

Comment (doanac) [reply client/foundries.go:1054]: force push fix

**Comment by doanac** [comment]
rebased on main and fixed ci-issue

**Pull request merged**